### PR TITLE
feat(docker-develop): adding mailhog service to docker-compose

### PR DESCRIPTION
**Pull request by jfernancordova**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/graycoreio/mage2docker/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[x] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[x] Other... Please describe: Adding service mailhog to docker-compose
```

## What is the current behavior?

Fixes: N

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

## Other information
Issue #11

### CODE REVIEW

**Comment by damienwebdev** [comment]
@jfernancordova this looks great, thanks for the contribution.  Could you write up some docs on usage?

**Comment by jfernancordova** [comment]
> @jfernancordova this looks great, thanks for the contribution. Could you write up some docs on usage?

Of course, I'm sorry for not writing one

**Comment by jfernancordova** [comment]
@damienwebdev I created a new folder called docker, and a new file in (docs/stories/docker/mailhog-service.md), I'm not sure if the correct path for this doc, let me know if it's ok.

File: docker-compose.yml [hunk 1]
```diff
@@ -16,6 +16,15 @@ services:
     networks:
       - frontend
       - backend
+  mailhog:
+    image: mailhog/mailhog:latest
+    ports: 
+      - 1025:1025
```
Comment (damienwebdev) [inline docker-compose.yml:22]: I don't think we should be exposing this on the host, as I don't think we need this from the host's perspective.

File: docker-compose.yml [hunk 2]
```diff
@@ -16,6 +16,15 @@ services:
     networks:
       - frontend
       - backend
+  mailhog:
```
Comment (damienwebdev) [inline docker-compose.yml:19]: Can we move the update from the root docker-compose into the `compose` folder?

File: docs/stories/docker/mailhog-service.md [hunk 3]
```diff
@@ -0,0 +1,14 @@
+# Mailhog service
+For development environments, the service mailhog helps to test emails locally, it's declared in the [docker-compose](../../../docker-compose.yml) and exports two ports, the first one is 1025 that it's used for smtp server, and the 8025 is a web interface, so you can see all test emails.
+
+For Inter-stack containers communication, this service can be used with its name and port, for example:
```
Comment (damienwebdev) [inline docs/stories/docker/mailhog-service.md:4]: I think it's possible to configure Magento via environment variables in this case

```
https://devdocs.magento.com/guides/v2.4/config-guide/deployment/pipeline/example/environment-variables.html#convert-configuration-paths-to-variable-names
```

Docs here: https://docs.magento.com/user-guide/system/email-communications.html

File: docs/stories/docker/mailhog-service.md [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+1. If mailhog sendmail is not installed in the php environments (magento's services):
+  ```php
+  wget https://github.com/mailhog/mhsendmail/releases/download/v0.2.0/mhsendmail_linux_amd64
+  sudo chmod +x mhsendmail_linux_amd64
+  sudo mv mhsendmail_linux_amd64 /usr/local/bin/mhsendmail
+  ```
+ 2. Change your php.ini sendmail_path value as follows:
+```php
+ sendmail_path = '/usr/local/bin/mhsendmail --smtp-addr="mailhog:1025"'
```
Comment (damienwebdev) [inline docs/stories/docker/mailhog-service.md:13]: I think we leave this out of `php.ini`

File: docs/stories/docker/mailhog-service.md [hunk 5]
```diff
@@ -0,0 +1,14 @@
+# Mailhog service
+For development environments, the service mailhog helps to test emails locally, it's declared in the [docker-compose](../../../docker-compose.yml) and exports two ports, the first one is 1025 that it's used for smtp server, and the 8025 is a web interface, so you can see all test emails.
+
+For Inter-stack containers communication, this service can be used with its name and port, for example:
+1. If mailhog sendmail is not installed in the php environments (magento's services):
+  ```php
+  wget https://github.com/mailhog/mhsendmail/releases/download/v0.2.0/mhsendmail_linux_amd64
```
Comment (damienwebdev) [inline docs/stories/docker/mailhog-service.md:7]: Hrm, I'm not familiar with Mailhog's install on linux, have a doc to review?

**Review by damienwebdev — CHANGES_REQUESTED** [review]

File: docker-compose.yml (hunk 1 above)
Comment (jfernancordova) [reply docker-compose.yml:22]: You're right, I'll remove it

File: docker-compose.yml (hunk 2 above)
Comment (jfernancordova) [reply docker-compose.yml:19]: Ok, no problem.

File: docs/stories/docker/mailhog-service.md (hunk 3 above)
Comment (jfernancordova) [reply docs/stories/docker/mailhog-service.md:4]: Perfect

File: docs/stories/docker/mailhog-service.md (hunk 4 above)
Comment (jfernancordova) [reply docs/stories/docker/mailhog-service.md:13]: Ok

File: docs/stories/docker/mailhog-service.md (hunk 5 above)
Comment (jfernancordova) [reply docs/stories/docker/mailhog-service.md:7]: https://github.com/mailhog/MailHog#debian--ubuntu

**Review by damienwebdev — APPROVED** [review]
lgtm

**Comment by damienwebdev** [comment]
@jfernancordova code looks good to me, please sign your commits with your gpg key!

**Comment by jfernancordova** [comment]
> @jfernancordova code looks good to me, please sign your commits with your gpg key!

Done, thank you so much!

**Comment by damienwebdev** [comment]
beautifully done, thanks so much!

**Pull request merged**